src/App.tsx

```diff
@@ -3,33 +3,49 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event.target.value,
+    });
+  };
 
   render() {
+    const visibleMovies = moviesFromServer.filter(({ title, description }) => (
+      title.toLowerCase().includes(this.state.query.toLowerCase())
+      || description.toLowerCase().includes(this.state.query.toLowerCase())
+    ));
```

Comment:
make variables for these. it's quite difficult to understand what is going on here